### PR TITLE
Support for Elasticsearch 2.x and 5.x

### LINKED ISSUE #4: Add support for elasticsearch 5.0.0
Currently even by specifying the version the upgrade does not work. The config format has changed a bit. for example transport.tcp.port is not recognized.

### PR DESCRIPTION
**Pull request by gaczm**
Besides:
- got rid of elasticsearch dependency
- few changes in API
- plugins are now installed using elasticsearch-plugin/plugin tool
- cluster name and tcp transport port are not set by default
- possibility to set ES_JAVA_OPTS

Closes #4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,13 +41,13 @@ And that's all, you can connect to your embedded-elastic instance on specified p
...
 | `withDownloadUrl(URL downloadUrl)` | if you prefer to download Elasticsearch from different location than official repositories you can do that using this method |
-| `withPortNumber(int portNumber)` | port number on which Elasticsearch will be started |
-| `withClusterName(String clusterName)` | cluster name for created Elasticsearch instance |
-| `withSetting(String name, Object value` | setting name and value as in elasticsearch.yml file |
-| `withMapping(InputStream mapping)`, `withMapping(String mapping)` | JSON with mapping of your index |
-| `withSettings(InputStream settings)`, `withSettings(String settings)` | JSON with settings of your index |
-| `withPlugin(String name, URL urlToDownload)` | plugin that should be installed into Elasticsearch; use multiple times for multiple plugins |
+| `withSetting(String key, Object value)` | setting name and value as in elasticsearch.yml file |
+| `withPlugin(String name, String expression)` | plugin that should be installed into Elasticsearch; treat expression as argument to `./elasticsearch-plugin install` command; use multiple times for multiple plugins |
```
Comment (pbobruk) [inline README.md]: withPlugin has only expression param

File: es50-test/src/test/groovy/pl/allegro/tech/embeddedelasticsearch/EmbeddedElasticSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+import org.elasticsearch.common.settings.Settings
+import org.elasticsearch.common.transport.InetSocketTransportAddress
+import org.elasticsearch.index.query.QueryBuilders
+import org.elasticsearch.transport.client.PreBuiltTransportClient
+import org.skyscreamer.jsonassert.JSONAssert
+import spock.lang.Shared
+import spock.lang.Specification
+
+import java.util.concurrent.TimeUnit
+
```
Comment (pbobruk) [inline es50-test/src/test/groovy/pl/allegro/tech/embeddedelasticsearch/EmbeddedElasticSpec.groovy]: unused imports

File: README.md (hunk 1 above)
Comment (pbobruk) [inline README.md]: I have found some unused imports in differed places, you can optimize imports.
Code looks cool. Good job!

Comment (gaczm) [inline README.md]: Ok, I've removed unused imports. So if you like the code I will merge this PR :)

**Pull request merged**